Sources/DistributedActors/Cluster/ClusterControl.swift

```diff
@@ -155,4 +155,77 @@ public struct ClusterControl {
     public func down(member: Cluster.Member) {
         self.ref.tell(.command(.downCommandMember(member)))
     }
+
+    public func joined(node: UniqueNode, within: Duration) async throws {
+        try await self.awaitMemberStatus(is: .up, within: within, node: node)
+    }
+
+    public func ensureNodes(_ status: Cluster.MemberStatus, within: Duration, nodes: [UniqueNode]) async throws {
+        try await withThrowingTaskGroup(of: Void.self) { group in
+            for node in nodes {
+                group.addTask {
+                    try await self.awaitMemberStatus(is: status, within: within, node: node)
+                }
+                try await group.next() // this is required to propagate error and cancel all tasks
```

Comment:
not required for cancelling; but yes required for the error propagation -- slight fixup in comment here only

Reply:
Slighly change this API perhaps, so the group actually waits concurrently:

```swift
            for node in nodes {
                 group.addTask {
                     try await self.awaitMemberStatus(is: status, within: within, node: node)
                 }
            }

// loop explicitly to propagagte any throws that might have happened 
            for try await status in group { nothing }
```

that try await on the group does what we need here

Reply:
oops, good catch